### PR TITLE
Send logs to client on fatal errors if possible

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Send logs to client on fatal errors if possible. This will make test results more readable.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
No tests will be provided for this feature.

**Comment by alexey-milovidov** [comment]
#11834

**Comment by alexey-milovidov** [comment]
> ClickHouse build check — 15/19 builds are OK 

> Build failed without build log. It's inner CI problem. Maybe something wrong with docker or host. Just wait for build restart.

It is an issue in infamous "Sandbox" infrastructure.

File: src/Core/Settings.h [hunk 1]
```diff
@@ -316,7 +316,7 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingBool, log_profile_events, true, "Log query performance statistics into the query_log and query_thread_log.", 0) \
     M(SettingBool, log_query_settings, true, "Log query settings into the query_log.", 0) \
     M(SettingBool, log_query_threads, true, "Log query threads into system.query_thread_log table. This setting have effect only when 'log_queries' is true.", 0) \
-    M(SettingLogsLevel, send_logs_level, LogsLevel::none, "Send server text logs with specified minimum level to client. Valid values: 'trace', 'debug', 'information', 'warning', 'error', 'none'", 0) \
+    M(SettingLogsLevel, send_logs_level, LogsLevel::fatal, "Send server text logs with specified minimum level to client. Valid values: 'trace', 'debug', 'information', 'warning', 'error', 'fatal', 'none'", 0) \
```
Comment (filimonov) [inline src/Core/Settings.h:319]: Compatibility?  Will it send value to old servers (which don't understand it)?

Comment (alexey-milovidov) [reply src/Core/Settings.h:319]: > Will it send value to old servers

No, because it is the default
(only explicit changes of settings are sent).

**Comment by alexey-milovidov** [comment]
This change will allow to stop run of functional test after first fatal error.

**Comment by alexey-milovidov** [comment]
After set `send_logs_level` to something other than `none` there are too many progress messages in client:

```
milovidov-desktop :) SELECT toDateTime64('2019-09-16 19:20:12', 0)

SELECT toDateTime64('2019-09-16 19:20:12', 0)

↖ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.) 
┌─toDateTime64('2019-09-16 19:20:12', 0)─┐
│                    2019-09-16 19:20:12 │
└────────────────────────────────────────┘
↑ Progress: 0.00 rows, 0.00 B (0.00 rows/s., 0.00 B/s.) 
↗ Progress: 1.00 rows, 1.00 B (767.04 rows/s., 767.04 B/s.) 

1 rows in set. Elapsed: 0.002 sec.
```

It's not a bug, technically speaking...

**Pull request merged**